CONTRIBUTING.md

```diff
@@ -0,0 +1,118 @@
+1. **Setup your environment**: Setup up your local environment to best interact with the code. Further information is given [below](#setup-the-environment).
+2. **Make atomic changes**: Changes should be atomic. As such, pull requests should contain only few commits, and each commit should only fix one issue or implement one feature, with a concise commit message.
+3. **Test your changes**: Test any changes locally for code quality and functionality and add new tests for any additional code. How to test is described [below](#test-changes).
+4. **Create semantic, conventional and signed commits**: Any commits should follow a simple semantic convention to help structure the work on Connaisseur. The convention is described [below](#semantic-and-conventional-commits). For security reasons and since integrity is at the core of this project, code merged into master must be signed. How we achieve this is described [below](#signed-commits-and-pull-requests).
+5. **Create Pull Request**: We consider code review central to quality and security of code. Therefore, a pull request (PR) should be created for each contribution. It will be reviewed, and potential improvements may be discussed within the PR. After approval, changes will be merged into the `master` branch.
+
+### Setup the Environment
+To start contributing, you will need to setup your local environment. First step is to get the source code by cloning this repository:
+```bash
+git clone git@github.com:sse-secure-systems/connaisseur.git
+```
+In order to review the effects of your changes, you should create your local Kubernetes cluster and install Connaisseur. This is described in the [setup guide](setup/README.md). A simple starting point may be a Minikube cluster with e.g. a [Docker Hub](https://hub.docker.com/) repository for maintaining your test images and trust data.
+
+### Test Changes
+Tests and linting are important to ensure code quality, functionality and security. We therefore aim to keep the code coverage high. We are running several automated tests in the [CI pipeline](https://github.com/sse-secure-systems/connaisseur/blob/master/.github/workflows/cicd.yaml). Application code is tested via [pytest](https://docs.pytest.org/) and linted via [pylint](https://pylint.org/). When making changes to the application code, please directly provide tests for your changes.
+
+We recommend using [black](https://pypi.org/project/black/) for autoformatting to simplify linting and reduce review effort. It can be installed via:
+```
+pip3 install black
+```
+To autoformat the code:
+```
+black <path-to-repository>/connaisseur
+```
+
+Changes can also be tested locally. We recommend the following approach for running pytest in a docker container:
+```
+docker run -it --rm -v <path-to-repository>/connaisseur:/data --entrypoint=ash python:alpine
+cd data
+pip3 install -r requirements.txt
+pip3 install .
+cd connaisseur
+pytest --cov=connaisseur --cov-report=xml tests/
+```
```

Comment:
Do changes in the application data propagate this way or does it require a container restart?